odl/discr/discr_mappings.py

```diff
@@ -442,9 +406,7 @@ def nearest(arg, out=None):
                 input_type = 'array'
 
             interpolator = _NearestInterpolator(
-                self.grid.coord_vectors,
-                x.asarray().reshape(self.grid.shape, order=self.order),
-                variant=self.variant,
+                self.grid.coord_vectors, x.asarray(), variant=self.variant,
```

Comment:
Actually, no changes required in the interpolators. Just removed the `.asarray()`